src/storage_engine.rs

```diff
@@ -68,7 +68,28 @@ impl StorageEngine {
         &mut self.db
     }
 
+    fn validate_table_options(&self, create_table: &CreateTableOptions) -> anyhow::Result<()> {
+        for (column, props) in create_table
+            .columns
+            .iter()
+            .filter(|(_, x)| x.foreign_key.is_some())
```

Comment:
I'm wondering if you could use a `filter_map` here instead to make this a little more concise:
```
.filter_map(|(_, x)| x.foreign_key.as_ref().map(|fk| (fk, x)))
```
And then you can move some of the nested if let some logic up in to the for loop call

Reply:
hmm I think this is kinda just moving the nesting to an extra set of method chains in a method which not sure if I prefer readability wise